### PR TITLE
Test 'Blob.compose' w/ 'user_project' set.

### PR DESCRIPTION
**Pull request by tseaver**

### CODE REVIEW

File: storage/tests/system.py [hunk 1]
```diff
@@ -547,6 +547,32 @@ def test_compose_replace_existing_blob(self):
         composed = original.download_as_string()
         self.assertEqual(composed, BEFORE + TO_APPEND)
 
+    @unittest.skipUnless(USER_PROJECT, 'USER_PROJECT not set in environment.')
```
Comment (dhermes) [inline storage/tests/system.py]: I'm beginning to think we should just have a `class TestWithUserProject(unittest.TestCase):` that has a `skipUnless` applied to the whole class.

File: storage/tests/system.py [hunk 2]
```diff
@@ -547,6 +547,32 @@ def test_compose_replace_existing_blob(self):
...
+    def test_compose_with_user_project(self):
+        new_bucket_name = 'compose-user-project' + unique_resource_id('-')
+        created = Config.CLIENT.create_bucket(
+            new_bucket_name, requester_pays=True)
+        try:
+            SOURCE_1 = b'AAA\n'
+            source_1 = created.blob('source-1')
```
Comment (dhermes) [inline storage/tests/system.py]: 1. The underscores seem to be superfluous?
1. Having `SOURCE_1` and `source_1` is a bad idea. `data1` and `blob1` are much better names.

File: storage/tests/system.py (hunk 1 above)
Comment (tseaver) [reply storage/tests/system.py]: We would have to replicate all the varioius `setUp` / `tearDown` bits I'm currently reusing.

File: storage/tests/system.py (hunk 2 above)
Comment (tseaver) [reply storage/tests/system.py]: It is a trivial copy of an existing test with the same names. Should I be widening the scope of this PR to address the other one?

Comment (dhermes) [reply storage/tests/system.py]: I'm not sure what you mean. I see one test case with not great variable names.

**Pull request merged**